src/core/ext/transport/chttp2/client/secure/secure_channel_create.c

```diff
@@ -60,69 +60,57 @@ static void client_channel_factory_unref(
 static grpc_subchannel_args *get_secure_naming_subchannel_args(
     grpc_exec_ctx *exec_ctx, const grpc_subchannel_args *args) {
   grpc_channel_credentials *channel_credentials =
-      grpc_find_channel_credentials_in_args(args->args);
-  if (channel_credentials == NULL) return NULL;
-
+      grpc_channel_credentials_find_in_args(args->args);
+  // Make sure security connector does not already exist in args.
+  if (grpc_security_connector_find_in_args(args->args) != NULL) {
+    gpr_log(GPR_ERROR, "Security connector already present in channel args.");
+    return NULL;
+  }
+  if (channel_credentials == NULL) {
+    gpr_log(GPR_ERROR,
+            "Can't create subchannel: channel credentials missing for secure "
+            "channel.");
+    return NULL;
+  }
   const grpc_slice_hash_table *targets_info =
       grpc_glb_targets_info_find_in_args(args->args);
-  if (targets_info == NULL) return NULL;
-
-  /* To which address are we connecting? */
+  // To which address are we connecting?
   const char *target_uri_str = grpc_get_subchannel_address_uri_arg(args->args);
```

Comment:
Suggest moving these lines into the "else" clause on line 87 below, since the result isn't used anywhere else.